### PR TITLE
[MC-43] Create logic component checkbox

### LINKED ISSUE #43: [Feature] Create Component Logic Checkbox
**Description**
I want a class to be created that represents the checkbox that will be used as the default for the entire site, being possible to instantiate the object for each component.

The following validations must be considered:
- Checkbox = valid/invalid, mandatory/mandatory

**Additional context**
const cbx1 = new Checkbox('btn query 1');
const cbx2 = new Checkbox('btn query 2');

**Acceptance Criteria**
- The component must have its class
- The component needs to keep its state (valid/invalid, disabled/enabled)
- The component must have a get/set value
- The component must have validation
- The component must have required (true|false)
- The component is expected to have tests

### PR DESCRIPTION
**Pull request by EvertonDuarteGuimaraes**
# Feature/Issue Description
- add getters/setters
- add click event
- add Component super class with enable/disabled method and required propriety
- add heritage for checkbox and dropdown
- change checkbox HTML
- change checkbox SCSS classes

# Changes Summary
- add checkbox component JS file
- add new Checkbox component class
- add checkbox component tests

# How Has This Been Tested?
- Run server
- Go to components, find checkbox.component.js file
- Go to main.js
- Use any checkbox methods available listed in CheckboxComponent class.

### CODE REVIEW

File: src/components/checkbox.component.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        this._addEventClickListner();
+    }
+
+    setLabel(text) {
+        this.text = text;
+        this._label.html(text);
+    }
+
+    setChecked() {
```
Comment (feelsantiago) [inline src/components/checkbox.component.js]: No need to add set prefix

File: src/components/checkbox.component.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        this._checkbox.prop('checked', true);
+        this.checked = this._checkbox.is(':checked');
+    }
+
+    setUnchecked() {
```
Comment (feelsantiago) [inline src/components/checkbox.component.js]: No need to set prefix

File: src/components/checkbox.component.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        this._checkbox.prop('checked', false);
+        this.checked = this._checkbox.is(':checked');
+    }
+
+    _addEventClickListner() {
+        this._checkbox.on('click', () => {
+            this.checked = this._checkbox.is(':checked');
```
Comment (feelsantiago) [inline src/components/checkbox.component.js]: This code is repeated 3 times.

**Review by feelsantiago — CHANGES_REQUESTED** [review]

File: src/components/checkbox.component.js (hunk 1 above)
Comment (EvertonDuarteGuimaraes) [reply src/components/checkbox.component.js]: Fixed

File: src/components/checkbox.component.js (hunk 2 above)
Comment (EvertonDuarteGuimaraes) [reply src/components/checkbox.component.js]: Fixed

File: src/components/checkbox.component.js (hunk 3 above)
Comment (EvertonDuarteGuimaraes) [reply src/components/checkbox.component.js]: Now it's a method

**Review by RodrigoSoaresNascimento — DISMISSED** [review]
Great!

**Review by tvps20 — DISMISSED** [review]

**Review by RodrigoSoaresNascimento — APPROVED** [review]

**Review by tvps20 — APPROVED** [review]

**Review by feelsantiago — APPROVED** [review]

**Pull request merged**